sycl/include/CL/sycl/accessor.hpp

```diff
@@ -669,9 +678,18 @@ class accessor :
   using AccessorSubscript =
       typename AccessorCommonT::template AccessorSubscript<Dims>;
 
-  using RefType = DataT &;
   using ConcreteASPtrType = typename detail::PtrValueType<DataT, AS>::type *;
+#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
+  using RefType =
+      typename std::conditional<AS == access::address_space::constant_space,
```

Comment:
This expression is used at least 6 times. Seems worth to factor out. Suggest change to:
```c++
//somewhere in traits:
#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
template<class AS, class DataT> using const_if_const_AS = typename std::conditional<AS == 
   access::address_space::constant_space, const DataT, DataT>::type;
#else
template<class AS, class DataT> using const_if_const_AS= DataT;
#endif
//and then here:
using RefType = const_if_const_AS<AS, DataT>&;
using PtrType = const_if_const_AS<AS, DataT>*;
```

Reply:
Sure, done.